x-pack/plugins/canvas/canvas_plugin_src/functions/common/compare.ts

```diff
@@ -3,14 +3,21 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { ContextFunctionSpec } from '../types';
 
-export function compare() {
+interface Arguments {
+  op: 'eq' | 'ne' | 'lt' | 'gt' | 'lte' | 'gte';
+  to: Context;
+}
+
+type Context = boolean | number | string | null;
```

Comment:
This occurs in eight places, likely worth abstracting out

Reply:
See above... `types` would get confusing, where you'd have a single type above, but the type wouldn't clear until you listed all of the possibilities in the `types` array.